airflow/bin/cli.py

```diff
@@ -185,6 +185,56 @@ def trigger_dag(args):
     logging.info(message)
 
 
+def batch_set_pools(args):
+    """
+    set a batch of Airflow pools
+
+    :param args: arguments from argparser
+    :return:
+    """
+    session = settings.Session()
+    pool_names = args.names
+    pool_slots = args.slots
+    pool_descriptions = args.descriptions
+    print("Number of pool names input: {}".format(len(pool_names)))
+    print("Number of pool slots count input: {}".format(len(pool_slots)))
+    print("Number of pool description input: {}".format(len(pool_descriptions)))
+
+    if len(pool_names) != len(pool_slots) or len(pool_names) != len(pool_descriptions):
+        print("Numbers of names, pools, descriptions are not equal to each other")
+    else:
+        pool_metadata_by_name = dict()
+        for i in range(len(pool_names)):
+            pool_metadata_by_name[pool_names[i]] = {'slots': pool_slots[i],
+                                                    'description': pool_descriptions[i]}
+        existing_pools = (
+            session.query(Pool)
+            .filter(Pool.pool.in_(pool_names))
+            .all())
+        for current_pool in existing_pools:
+            pool_metadata = pool_metadata_by_name.get(current_pool.pool)
+            input_pool_slots = pool_metadata.get('slots')
+            input_pool_desc = pool_metadata.get('description')
+            if int(input_pool_slots) != int(current_pool.slots) or input_pool_desc != current_pool.description:
+                print("Need to update pool: {}".format(current_pool.pool))
+                print("Original slots: {}\nNew slots:{}".format(current_pool.slots, input_pool_slots))
+                print("Original description: {}\nNew description:{}".format(current_pool.description, input_pool_desc))
+                current_pool.slots = input_pool_slots
+                current_pool.description = input_pool_desc
+            else:
+                print("No need to update pool: {}".format(current_pool.pool))
+            del pool_metadata_by_name[current_pool.pool]
```

Comment:
it might be nicer to use `pool_metadata_by_name.pop(current_pool.pool)` instead of `del`

Reply:
https://stackoverflow.com/questions/5713218/best-method-to-delete-an-item-from-a-dict Since we don't need the removed item, using del or pop makes no difference?

Reply:
earggghhhh. I think del is less pythonic, but the internet seems to not really have an opinion, so I guess it's fine. 
(I generally don't love the keywords in python -> like how they moved from `print hello` to `print(hello)`